src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,129 @@
+import classNames from 'classnames';
+import { useState } from 'react';
+import { Todo } from '../../types/Todo';
+
+type TodoItemProps = {
+  todo: Todo;
+  loadingId: number[];
+  removeTodo: (todoId: number) => void;
+  changeTodo: (
+    property: string,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    value: any,
+    todoId: number,
+  ) => void;
+};
+
+export const TodoItem: React.FC<TodoItemProps> = ({
+  todo: { id, title, completed },
+  loadingId,
+  removeTodo,
+  changeTodo,
+}) => {
+  const [isEditingId, setIsEditingId] = useState<number | null>(null);
+  const [inputValue, setInputValue] = useState(title);
+
+  const handleDoubleClick = () => {
```

Comment:
you should also focus on input when it appears - use ref for it (ask in chat if you have questions)